src/core/model/types/double_type.h

```diff
@@ -9,10 +11,16 @@ class DoubleType final : public NumericType<Double> {
     DoubleType() noexcept : NumericType<Double>(TypeId::kDouble) {}
 
     CompareResult Compare(std::byte const* l, std::byte const* r) const final {
+        return CompareULPs(l, r, 5);  // eps = 3 is too low for comparing doubles calculated using
+                                      // arithmetic operations
+    }
+
+    CompareResult CompareULPs(std::byte const* l, std::byte const* r, int eps) const {
```

Comment:
Why is it named `eps`? It's not an epsilon in any way, because of this `eps * std::numeric_limits<model::Double>::epsilon()` looks like we're multiplying epsilon values which is really strange

Reply:
Also let's replace `int` with `unsigned int` since this value is not expected to be negative